src/db/connection.cr

```diff
@@ -42,10 +44,60 @@ module DB
     # :nodoc:
     abstract def build_unprepared_statement(query) : Statement
 
+    def begin_transaction
+      raise DB::Error.new("There is an existing transaction in this connection") if @transaction
+      @transaction = true
+      create_transaction
+    end
+
+    protected def create_transaction : Transaction
+      TopLevelTransaction.new(self)
+    end
+
     protected def do_close
       @statements_cache.each_value &.close
       @statements_cache.clear
       @database.pool.delete self
     end
```

Comment:
Is it safe to close a connection without rolling back its transactions? I assume it is, but just checking.